tt_metal/hw/inc/debug/sanitize_noc.h

```diff
@@ -51,17 +49,94 @@ typedef bool debug_sanitize_noc_dir_t;
 #define DEBUG_SANITIZE_NOC_UNICAST false
 typedef bool debug_sanitize_noc_cast_t;
 
+// Helper function to get the core type from noc coords.
+int get_core_type(uint8_t noc_id, uint32_t x, uint32_t y) {
+    core_info_msg_t tt_l1_ptr *core_info = GET_MAILBOX_ADDRESS_DEV(core_info);
+
+    for (uint32_t idx = 0; idx < MAX_DRAM_CORES; idx++) {
```

Comment:
A couple thoughts:
1) Seems cores can be uint8, right?  That would save a fair amount of space
2) Making dram/eth/pcie work by storing X,Y,type w/ 1 loop and no checks for "invalid" would save a bunch of code space and likely be more performant (use invalid X,Y to prevent a match).  Thoughts?